docs/maven-plugin.md

```diff
@@ -0,0 +1,222 @@
+| `binaryStaticContent`      | If to generate a [binary content](binary-rendering.md) resource for each template | `false`                                                        |
+| `compileArgs`              | Sets additional compiler arguments for `jte` templates.                           | See note below                                                 |
+| `kotlinCompileArgs`        | Sets additional compiler arguments for `kte` templates                            | None                                                           |
+| `packageName`              | The package name, where template classes are generated to                         | [`Constants.PACKAGE_NAME_PRECOMPILED`][constants-package-name] |
+| `keepGeneratedSourceFiles` | If it should keep all generated jte source files after compilation                | `false`                                                        |
+
+!!! info "About `htmlPolicyClass`"
+
+    The `htmlPolicyClass` will default to `gg.jte.html.OwaspHtmlPolicy` if the content type is `Html`.
+
+!!! info "About `compileArgs`"
+
+    This option default depends on the `maven.compiler.release` or, if it is absent, on `maven.compiler.source` and `maven.compiler.target` properties.
+
+    So, if the following property is configured in your `pom.xml`:
+
+    ```xml linenums="1"
+    <properties>
+        <maven.compiler.release>21</maven.compiler.release>
+    </properties>
+    ```
+    
+    The `compileArgs` will default to `--release 21`. If `maven.compiler.release` is not configured, it will try to use `maven.compiler.source` and `maven.compiler.target` instead:
+
+    ```xml linenums="1"
+    <properties>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
+    </properties>
+    ```
+
+    This will result in `--source 21 --target 21` as the default value for `compileArgs`. If none of those properties are configured, then the default value will be empty. Release is usually preferred over source and target since it will also ensure the correct API versions are used. See more details in the links below:
+
+    - [Maven Compiler Plugin](https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#parameter-details)
+    - [`-release` option](https://docs.oracle.com/en/java/javase/21/docs/specs/man/javac.html#option-release)
+    - [`-source`](https://docs.oracle.com/en/java/javase/21/docs/specs/man/javac.html#option-source) and [`-target`](https://docs.oracle.com/en/java/javase/21/docs/specs/man/javac.html#option-target) options
+
+!!! warning "Clean all before precompiling"
+
+    The `precompile` goal cleans the directory containing the compiled template classes every time it runs. In your local development environment, it may make more sense to use [hot reloading](hot-reloading.md).
```

Comment:
The hint to hot reloading seems unrelated to the `Clean all before precompiling` warning.